### ISSUE TITLE
segfault trying to call clean_contents on certain pdfs with python 3.12

### ISSUE THREAD

**Issue by Luux** [pymupdf/PyMuPDF#2907]
### Description of the bug

Trying to use `clean_contents` on certain pdf files causes a segmentation fault when using Python 3.12. Interestingly, the same setup on Python 3.11 works just fine.
I managed to strip one of the pdfs down with `mutool trim` as far as possible to reproduce the bug. Trimming it further results in the segfault to disappear.

### How to reproduce the bug

```
import pathlib
import fitz


def test_segfault():
    pdf_file = pathlib.Path("test11.pdf").read_bytes()
    fitz_document = fitz.open(stream=pdf_file, filetype="application/pdf")

    pdf_pages = list(fitz_document.pages())
    (page,) = pdf_pages
    page.clean_contents()

test_segfault()
```
[test11.pdf](https://github.com/pymupdf/PyMuPDF/files/13704140/test11.pdf)

### PyMuPDF version

1.23.7

### Operating system

Linux

### Python version

3.12

**Comment by julian-smith-artifex-com** [comment]
Thanks for the clear and simple reproducer. I've reproduced the problem.

Interestingly, the new "rebased" implementation of PyMuPDF appears to work fine here. The rebased implementation is included in current releases and is a drop-in replacement for the classic implementation, so i recommend you use it yourself here - just change `import fitz` to `import fitz_new as fitz`. See #2680 for more information.

The rebased implementation will become the default (with `import fitz`) quite soon after this week's release of 1.23.8.

Seeing as rebased works here, we do not intend to fix this issue in the classic implementation. Of course please let us know if you have any problems with `import fitz_new as fitz`.

**Comment by Luux** [comment]
I can confirm that it works with `fitz_new`

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.9 where `import fitz` gets the rebased implementation.
